src/app/pages/send/send.component.tsx

```diff
@@ -227,6 +233,32 @@ export class SendPage extends Component<IProps, IState> {
         );
     }
 
+    public getAddressFromName(name) {
+        let address: string = name;
+
+        if (name.charAt(name.length - 4) === '.') {
+            const nameService = name.substring(name.length - 3);
+            const namicorn = new Namicorn({ debug: true, disableMatcher: true });
+            if (nameService === 'eth') {
+                namicorn.use(namicorn.middleware.ens());
+            } else if (nameService === 'zil') {
+                namicorn.use(namicorn.middleware.zns());
+            } else if (nameService === 'rsk') {
+                namicorn.use(namicorn.middleware.rns());
+            }
+            address = namicorn
+                .resolve(name)
+                .then(x => {
+                    return x;
+                })
+                .catch(error => {
+                    return name;
+                });
+        }
+
+        return address;
```

Comment:
Ok, so `namicorn` seems to be `async` so, maybe we should always return a `Promise` instead, and on the change handler wait for it to resolve.